File: della_parser_bot/parser/url_generator.js
// Code from della.kz

function make_short_search_query_params() {
    var location = '',
    truck_array = [
    ],
    weightFrom = document.getElementById('SelectWeightFrom') && document.getElementById('SelectWeightFrom').value || 0,
    weightTo = document.getElementById('SelectWeightTo') && document.getElementById('SelectWeightTo').value || 0,
    cubeFrom = document.getElementById('SelectCubeFrom') && document.getElementById('SelectCubeFrom').value || 0,
    cubeTo = document.getElementById('SelectCubeTo') && document.getElementById('SelectCubeTo').value || 0,
    date_from = document.getElementById('calInputTextFrom') && document.getElementById('calInputTextFrom').getAttribute('datevalue').replace(/\//g, '-') || 0,
    date_to = document.getElementById('calInputTextTo') && document.getElementById('calInputTextTo').getAttribute('datevalue').replace(/\//g, '-') || 0,
    cargo = document.getElementById('classicCargo_').checked ? 'cargo' : 'trans',
    cid = !1,
    rid = !1,
    citid = !1,
    cf_array = [
    ],
    rf_array = [
    ],
    cif_array = [
    ],
    ct_array = [
    ],
    rt_array = [
    ],
    cit_array = [
    ],
    got_cities_from = !1,
    got_cities_to = !1;
    if ($('.multiTruckBlock .truck_select').each((function () {
        for (var truck_id = 6 != $(this).val() && $(this).val() || 0, unique = !0, i = 0; i < truck_array.length; i++) if (truck_array[i] == truck_id) {
            unique = !1;
            break
        }
        unique && truck_array.push(truck_id)
    })), $('#marshrutteKeeperFrom table.multisearchBlock').each((function () {
        cf_array.length < 8 && (cid = $(this).find('.country_select').val(), rid = $(this).find('.region_select').val(), citid = $(this).find('.city_input').attr('data-city-uid'), '0' == cid || cf_array.slice( - 1) [0] == cid && rf_array.slice( - 1) [0] == rid && cif_array.slice( - 1) [0] == citid || (cf_array.push(cid), rf_array.push(rid), cif_array.push(citid), 0 != citid && (got_cities_from = !0)))
    })), $('#marshrutteKeeperTo table.multisearchBlock').each((function () {
        ct_array.length < 8 && (cid = $(this).find('.country_select').val(), rid = $(this).find('.region_select').val(), citid = $(this).find('.city_input').attr('data-city-uid'), '0' == cid || ct_array.slice( - 1) [0] == cid && rt_array.slice( - 1) [0] == rid && cit_array.slice( - 1) [0] == citid || (ct_array.push(cid), rt_array.push(rid), cit_array.push(citid), 0 != citid && (got_cities_to = !0)))
    })), 1 == rf_array.length && '0' == rf_array[0] && (rf_array = [
    ]), 1 == rt_array.length && '0' == rt_array[0] && (rt_array = [
    ]), got_cities_from || (cif_array = [
    ]), got_cities_to || (cit_array = [
    ]), !cf_array.length && !ct_array.length) return alert(g_phrase[1030]),
    !1;
    if ($('#search_request_button').css('color', 'red'), $('#t44k').css('color', 'red'), location += 'a' + cf_array.join('l') + 'b' + rf_array.join('l') + (cif_array.length ? 'j' + cif_array.join('l') : '') + 'd' + ct_array.join('l') + 'e' + rt_array.join('l') + (cit_array.length ? 't' + cit_array.join('l') : ''), location += 'f', 0 === cubeFrom && 0 === cubeTo || (0 != cubeFrom && (location += cubeFrom), location += 'l', 0 != cubeTo && (location += cubeTo)), location += 'o', 0 === weightFrom && 0 === weightTo || (0 != weightFrom && (location += weightFrom), location += 'l', 0 != weightTo && (location += weightTo)), [
        {
            mask: 'z1',
            selector: '.isNotDeleted_filter'
        },
        {
            mask: 'z2',
            selector: '.isCustomer_filter',
            additionalCondition: function ($filterContainer) {
                return 'cargo' === cargo
            }
        },
        {
            mask: 'z3',
            selector: '.isPartner_filter'
        },
        {
            mask: 'z4',
            selector: '.isNotBlack_filter'
        },
        {
            mask: 'z9',
            selector: '.isLoadUp_filter'
        },
        {
            mask: 'y1',
            selector: '.isADR_filter'
        },
        {
            mask: 'y3',
            selector: '.isTransWithGPS_filter',
            additionalCondition: function ($filterContainer) {
                return 'trans' === cargo
            }
        },
        {
            mask: 'y4',
            selector: '.isHumanitarian_filter'
        },
        {
            mask: 'z7',
            selector: '.isOnlyNew_filter'
        },
        {
            mask: 'y5',
            selector: '.isOnlyFromVerified_filter'
        },
        {
            mask: 'y6',
            selector: '.groupFilter_filter',
            generateValue: $filterContainer=>{
                let value = Number($filterContainer.find('#group_filter_select option:selected').val());
                return isNaN(value) ? 0 : value
            }
        },
        {
            mask: 'z5',
            selector: '.priceType_filter',
            generateValue: $filterContainer=>{
                let value = 0;
                return $filterContainer.find('.filter_checkbox_tag.checked').each((function () {
                    let tagMask = $(this).attr('data-tag-mask');
                    value |= tagMask
                })),
                value
            }
        },
        {
            mask: 'z6',
            selector: '#search_time_filter_select',
            generateValue: $filterContainer=>{
                let value = Number($filterContainer.find('option:selected').attr('data-time-mask'));
                return isNaN(value) ? 0 : value
            }
        },
        {
            mask: 'y2',
            selector: '.loadType_filter',
            generateValue: $filterContainer=>{
                let value = 0;
                return $filterContainer.find('.filter_checkbox_tag.checked').each((function () {
                    let tagMask = $(this).attr('data-tag-mask');
                    value |= tagMask
                })),
                value
            }
        }
    ].forEach((function (filter) {
        let value,
        $filterContainer = $(filter.selector);
        value = 'function' == typeof filter.generateValue ? filter.generateValue($filterContainer) : Number($filterContainer.find('.filter_checkbox').hasClass('checked'));
        let additionalCondition = !0;
        'function' == typeof filter.additionalCondition && (additionalCondition = filter.additionalCondition($filterContainer)),
        0 !== value && additionalCondition && (location += filter.mask + value)
    })), location += 'h' + truck_array.join('l'), location += 'i', 0 !== date_from || 0 !== date_to) {
        var timeNow = new Date,
        timeFrom = getDateObject(date_from),
        timeTo = getDateObject(date_to),
        is_add_date_params = !0;
        timeNow.getDate() == timeFrom.getDate() && timeNow.getMonth() == timeFrom.getMonth() && timeNow.getYear() == timeFrom.getYear() && (timeNow.getDate() == timeTo.getDate() && timeTo.getMonth() - timeNow.getMonth() == 2 && timeNow.getYear() == timeTo.getYear() || timeNow.getDate() == timeTo.getDate() && timeTo.getMonth() + 12 - timeNow.getMonth() == 2 && timeNow.getYear() == timeTo.getYear() - 1) && (is_add_date_params = !1),
        location += is_add_date_params ? convertDate(date_from) + 'l' + convertDate(date_to) : 'l'
    }
    return location += 'k' + ('trans' == cargo ? 1 : 0),
    location += 'm' + (document.getElementById('bookmark_number') ? document.getElementById('bookmark_number').value : '')
}
function make_short_search_query_url() {
    var params = make_short_search_query_params();
    return !!params && '//' + DELLA_URL + ADD_PATH + '/search/' + params + '.html'
}

File: della_parser_bot/parser/url_generator.py
def make_short_search_query_params(
    cal_input_text_from: str,
    cal_input_text_to: str,
    classic_cargo: bool,
    truck_select: list[str],
    multisearch_block_from: list[list[str]],
    multisearch_block_to: list[list[str]],
    is_not_deleted_filter: bool,
    is_customer_filter: bool,
    is_partner_filter: bool,
    is_not_black_filter: bool,
    is_load_up_filter: bool,
    is_adr_filter: bool,
    is_trans_with_gps_filter: bool,
    is_humanitarian_filter: bool,
    is_only_new_filter: bool,
    is_only_from_verified_filter: bool,
    group_filter_select: str,
    price_type_filter: list[str],
    search_time_filter_select: str,
    load_type_filter: list[str],
    bookmark_number: str,
    weight_from: str = '0',
    weight_to: str = '0',
    cube_from: str = '0',
    cube_to: str = '0',
) -> str:
    location = ''
    truck_array = []
    date_from = (
        cal_input_text_from.replace('/', '-') if cal_input_text_from else '0'
    )
    date_to = cal_input_text_to.replace('/', '-') if cal_input_text_to else '0'
    cargo = 'cargo' if classic_cargo else 'trans'
    citid = (
        cf_array
    ) = rf_array = cif_array = ct_array = rt_array = cit_array = []
    False

    for truck_id in truck_select:
        if truck_id != '6' and truck_id not in truck_array:
            truck_array.append(truck_id)

    for multisearch_block in multisearch_block_from:
        if len(cf_array) < 8:
            cid = multisearch_block[0]
            rid = multisearch_block[1]
            citid = multisearch_block[2]
            if cid == '0' or (
                len(cf_array) > 0
                and cf_array[-1] == cid
                and rf_array[-1] == rid
                and cif_array[-1] == citid
            ):
                continue
            cf_array.append(cid)
            rf_array.append(rid)
            cif_array.append(citid)
            if citid != '0':
                pass

    for multisearch_block in multisearch_block_to:
        if len(ct_array) < 8:
            cid = multisearch_block[0]
            rid = multisearch_block[1]
            citid = multisearch_block[2]
            if cid == '0' or (
                len(ct_array) > 0
                and ct_array[-1] == cid
                and rt_array[-1] == rid
                and cit_array[-1] == citid
            ):
                continue
            ct_array.append(cid)
            rt_array.append(rid)
            cit_array.append(citid)
            if citid != '0':
                pass

    if not cf_array and not ct_array:
        raise ValueError('No search location provided')

    location += f'a{"l".join(cf_array)}b{"l".join(rf_array)}'
    if cif_array:
        location += f'j{"l".join(cif_array)}'
    location += f'd{"l".join(ct_array)}e{"l".join(rt_array)}'
    if cit_array:
        location += f't{"l".join(cit_array)}'
    location += 'f'

    if cube_from != '0' or cube_to != '0':
        if cube_from != '0':
            location += cube_from
        location += 'l'
        if cube_to != '0':
            location += cube_to
    location += 'o'

    if weight_from != '0' or weight_to != '0':
        if weight_from != '0':
            location += weight_from
        location += 'l'
        if weight_to != '0':
            location += weight_to

    filters = [
        {'mask': 'z1', 'condition': is_not_deleted_filter},
        {'mask': 'z2', 'condition': is_customer_filter and cargo == 'cargo'},
        {'mask': 'z3', 'condition': is_partner_filter},
        {'mask': 'z4', 'condition': is_not_black_filter},
        {'mask': 'z9', 'condition': is_load_up_filter},
        {'mask': 'y1', 'condition': is_adr_filter},
        {
            'mask': 'y3',
            'condition': is_trans_with_gps_filter and cargo == 'trans',
        },
        {'mask': 'y4', 'condition': is_humanitarian_filter},
        {'mask': 'z7', 'condition': is_only_new_filter},
        {'mask': 'y5', 'condition': is_only_from_verified_filter},
        {'mask': 'y6', 'value': int(group_filter_select)},
        {'mask': 'z5', 'value': sum(map(int, price_type_filter))},
        {'mask': 'z6', 'value': int(search_time_filter_select)},
        {'mask': 'y2', 'value': sum(map(int, load_type_filter))},
    ]

    for filt in filters:
        if filt.get('condition'):
            location += f'{filt["mask"]}{filt.get("value", 1)}'

    location += f'h{"l".join(truck_array)}'

    if date_from != '0' or date_to != '0':
        timeNow = datetime.now()
        time_from = datetime.strptime(date_from, '%Y-%m-%d')
        time_to = datetime.strptime(date_to, '%Y-%m-%d')

        is_add_date_params = True
        if (
            timeNow.day == time_from.day
            and timeNow.month == time_from.month
            and timeNow.year == time_from.year
            and (
                (
                    timeNow.day == time_to.day
                    and time_to.month - timeNow.month == 2
                    and timeNow.year == time_to.year
                )
                or (
                    timeNow.day == time_to.day
                    and time_to.month + 12 - timeNow.month == 2
                    and timeNow.year == time_to.year - 1
                )
            )
        ):
            is_add_date_params = False

        if is_add_date_params:
            location += f'{date_from}l{date_to}'
        else:
            location += 'l'

    location += f'k{int(cargo == "trans")}'
    location += f'm{bookmark_number}' if bookmark_number else ''

    return location


def make_short_search_query_url() -> str:
    params = make_short_search_query_params(
        ...
    )  # pass in all the required arguments
    return f'//{DELLA_URL}{ADD_PATH}/search/{params}.html' if params else ''
